packages/components/nodes/utilities/CustomFunction/CustomFunction.ts

```diff
@@ -123,14 +128,20 @@ class CustomFunction_Utilities implements INode {
         const vm = new NodeVM(nodeVMOptions)
         try {
             const response = await vm.run(`module.exports = async function() {${javascriptFunction}}()`, __dirname)
-            if (typeof response === 'string') {
+            const isEndingNode = Object.values(nodeData.outputs ?? {}).includes('EndingNode')
```

Comment:
To solve the double run issue, we can move the isEndingNode logic to the top of the init function and then reuse the variable down here:
```
  const isEndingNode = nodeData?.outputs?.output === 'EndingNode'
  if (isEndingNode && !options.isRun) return

```